test/grpc/leaderboard_test.dart

```diff
@@ -0,0 +1,32 @@
+import 'package:faker/faker.dart';
+import 'package:nakama/api.dart' as api;
+import 'package:nakama/nakama.dart';
+import 'package:test/expect.dart';
+import 'package:test/scaffolding.dart';
+
+import '../config.dart';
+
+void main() {
+  group('[gRPC] Test Leaderboard', () {
+    late final NakamaBaseClient client;
+    late final Session session;
+
+    setUpAll(() async {
+      client = getNakamaClient(
+        host: kTestHost,
+        ssl: false,
+        serverKey: kTestServerKey,
+      );
+
+      session = await client.authenticateDevice(deviceId: faker.guid.guid());
+    });
+
+    test('list leaderboard records', () async {
+      final result = await client.listLeaderboardRecords(
+          session: session, leaderboardId: 'test');
+
+      expect(result, isA<api.LeaderboardRecordList>());
+      expect(result.records, hasLength(0));
+    });
+  });
+}
```

Comment:
The tests assume a leaderboard with name `test` is created, which can't be created from client side, I don't know if this already exists or not, but maybe there can be a simple docker compose file with server code to prepare these.